pkg/local_object_storage/metabase/version.go

```diff
@@ -105,3 +110,50 @@ func migrateFrom2Version(db *DB, tx *bbolt.Tx) error {
 
 	return updateVersion(tx, 3)
 }
+
+func migrateFrom3Version(_ *DB, tx *bbolt.Tx) error {
+	c := tx.Cursor()
+	pref := []byte{metadataPrefix}
+	if k, _ := c.Seek(pref); bytes.HasPrefix(k, pref) {
+		return fmt.Errorf("key with prefix 0x%X detected", pref)
+	}
+	err := tx.ForEach(func(name []byte, b *bbolt.Bucket) error {
+		switch name[0] {
+		default:
+			return nil
+		case primaryPrefix, tombstonePrefix, storageGroupPrefix, lockersPrefix, linkObjectsPrefix:
+		}
+		if len(name[1:]) != cid.Size {
+			return fmt.Errorf("invalid container bucket with prefix 0x%X: wrong CID len %d", name[0], len(name[1:]))
+		}
+		cnr := cid.ID(name[1:])
+		err := b.ForEach(func(k, v []byte) error {
+			if len(k) != oid.Size {
+				return fmt.Errorf("wrong OID key len %d", len(k))
+			}
+			id := oid.ID(k)
+			var hdr object.Object
+			if err := hdr.Unmarshal(v); err != nil {
+				return fmt.Errorf("decode header of object %s from bucket value: %w", id, err)
+			}
+			par := hdr.Parent()
+			if err := putMetadataForObject(tx, hdr, par == nil, true); err != nil {
```

Comment:
i am always wondering what is `root && phy == true` and seems like this code _may_ accept it (and `putMetadataForObject` in general)

Reply:
mb `false`? `true` is a singular user object (small)

Reply:
hm, always had problems with fixing these terms' meanings. well if `root == true` is ok for small objects (and it seems we explain it [unequivocally](https://github.com/nspcc-dev/neofs-api/blob/00bfea07e771ee57f267d2a6387705f300057fea/object/types.proto#L111-L119)) then ok